### DISCUSSION TITLE
Helix-templete-generator

### DISCUSSION THREAD

**Discussion by littlelevi** [helix-editor/helix#2382] [Show and tell]
In reference to: #2348 

I was a bit frustrated with Helix not forgiving mistakes in theme files. After couple of minutes for searching a bug, I decided that I will write my own theme generator. That's how HTG was born. Helix Theme Generator, HTG for short, spits out error free(hopefully ) theme. Courtesy of the C ++ language, it makes sure it is always correct, checking for some bugs in compile time. Plus it allows to use features that I originally missed, about which I wrote in the above-mentioned discussion. It C++ may not be the most convenient tool for this job, but it does it pretty well.

https://github.com/littlelevi/htg

I do not count on any wide adoption, I treat it rather as a proof of concept of what I wrote about in other thread. Maybe a few people will find it useful it or will inspire them to create an even better tool or inspire them to introduce such features directly into the helix.

I don't expect to maintain htg. It was created over the weekend and its lifetime will be until Helix gets a plug-in system or at least a better theme parser.

**Comment by archseer** [comment]
There's also this one: https://github.com/CptPotato/helix-themes
